src/smartHandler.ts

```diff
@@ -195,10 +202,23 @@ export class SMARTHandler implements Authorization {
 
         if (fhirUserObject) {
             const { hostname, resourceType, id } = fhirUserObject;
+
             if (isFhirUserAdmin(fhirUserObject, this.adminAccessTypes, fhirServiceBaseUrl)) {
                 // if an admin do not add limiting search filters
                 return [];
             }
+
+            references.add(`${hostname}/${resourceType}/${id}`);
+            if (hostname === fhirServiceBaseUrl) {
+                references.add(`${resourceType}/${id}`);
+            }
+            if (request.resourceType && request.resourceType === resourceType) {
+                ids.add(id);
+            }
+        }
+
+        if (patientOrgsClaim) {
+            const { hostname, resourceType, id } = patientOrgsClaim;
             references.add(`${hostname}/${resourceType}/${id}`);
```

Comment:
We should decide whether the claim should carry the full URL of the patient organization. Currently we aren't including it, and this code doesn't take into account that it wasn't provided.

Reply:
[getFhirResource](https://github.com/awslabs/fhir-works-on-aws-authz-smart/blob/mainline/src/smartAuthorizationHelper.ts#L29) function returns FhirResource that takes in the claims and adds the baseurl.

Reply:
Got it. Thank you.